### PR TITLE
Added invalid cidr/ip tests for floating_ip_add and subnet_add

### PR DESCRIPTION
**Pull request by mmojzis**
Also added needed views and create function.

### CODE REVIEW

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 45c0bf4ff3118c7978c8b743f8efe888992cab76

The *local* fixture **`network_manager`** is used in the following files:
- **cfme/tests/networks/test_sdn_subnets.py**
    - *test_network_subnet_invalid_cidr*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 79d83441e692d8ec76dc2c26323d130fafa00103

The *local* fixture **`network_manager`** is used in the following files:
- **cfme/tests/networks/test_sdn_subnets.py**
    - *test_network_subnet_invalid_cidr*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/networks/test_sdn_addresses.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    test_requirements.sdn,
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([OpenStackProvider], scope="module")
+]
+
+
+@pytest.fixture()
+def network_manager(appliance, provider):
+    network_managers = appliance.collections.network_providers.filter({"provider": provider})
+    yield network_managers.all()[0]
```
Comment (jaryn) [inline cfme/tests/networks/test_sdn_addresses.py:17]: This is bit suspicious to me. How is it ensured the first returned network_manager is the one the test wants?

File: cfme/tests/networks/test_sdn_addresses.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+pytestmark = [
+    test_requirements.sdn,
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([OpenStackProvider], scope="module")
+]
+
+
+@pytest.fixture()
+def network_manager(appliance, provider):
+    network_managers = appliance.collections.network_providers.filter({"provider": provider})
```
Comment (jaryn) [inline cfme/tests/networks/test_sdn_addresses.py:16]:
```suggestion
    network_manager, = appliance.collections.network_providers.filter({"provider": provider}).all()
    yield network_manager
```

This way, when there are more than one network_provider, we will get an error. This is a good thing as it will notify us that something is not as we did expect.

```
In [6]: a, = (b for b in range(1))                                                            

In [7]: a                                                                                     
Out[7]: 0

In [8]: a, = (b for b in range(2))                                                            
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-c3f22dcf86ee> in <module>
----> 1 a, = (b for b in range(2))

ValueError: too many values to unpack (expected 1)
```

File: cfme/tests/networks/test_sdn_subnets.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    with pytest.raises(AssertionError):
+        subnets_collection.create(network_manager=network_manager.name,
+                                  network_name=provider.data['public_network'],
+                                  tenant=provider.data['tenant'], cidr=invalid_cidr,
+                                  name='test_subnet', provider=provider)
+
+    view = subnets_collection.create_view(SubnetView)
+    view.flash.assert_message(
+        "Unable to create Cloud Subnet: Invalid input for cidr. Reason: '{}' is not a valid IP "
+        "subnet.".format(invalid_cidr))
```
[the remark this answers is not in the document]
Comment (mmojzis) [reply cfme/tests/networks/test_sdn_subnets.py:42]: What is f-string?

File: cfme/tests/networks/test_sdn_addresses.py (hunk 1 above)
Comment (mmojzis) [reply cfme/tests/networks/test_sdn_addresses.py:17]: Well, how can openstack provider have multiple network managers? Never seen this option in CFME.
Also we are using the same fixture for storage managers and it works well. : https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/storage/test_manager.py#L31

File: cfme/tests/networks/test_sdn_subnets.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+pytestmark = [
+    test_requirements.sdn,
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([OpenStackProvider], scope="module")
+]
+
+
+@pytest.fixture()
+def network_manager(appliance, provider):
```
[the remark this answers is not in the document]
Comment (mmojzis) [reply cfme/tests/networks/test_sdn_subnets.py:15]: Well, in the future when they add another providers with network managers the fixture makes sense.

File: cfme/tests/networks/test_sdn_addresses.py (hunk 2 above)
Comment (mmojzis) [reply cfme/tests/networks/test_sdn_addresses.py:16]: Sorry, I don't understand the code.

File: cfme/tests/networks/test_sdn_subnets.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jaryn) [reply cfme/tests/networks/test_sdn_subnets.py:42]: This is f-string.
```
my_identifier = "baz"
f"foo {my_identifier} bar"
``

File: cfme/tests/networks/test_sdn_addresses.py (hunk 2 above)
Comment (jaryn) [reply cfme/tests/networks/test_sdn_addresses.py:16]: The code is using Pythons value unpacking to make sure the value we get is only one in the iterable. 

You can do this with tuples:
```
a, b, c = 1, 2, 3
```
which assigns `a` with 1 and `b` with 2, `c` with 3

Of course ths works with single-item tuples:
```
a, = [1]
```
this assings `a` with 1. But it has the advantage of failing when more items are being unpacked:

```
a, = 1,2
```
will fail with `ValueError: too many values to unpack (expected 1)`

This works for any iterable, including generators. So when you are tempting to do
```my_variable = some_iterable[0]```
and you assume only single item is available in `some_variable`, it is good to do
```my_variable, = some_iterable```
instead, because it will fail when the assumption is not met.

Comment (mmojzis) [reply cfme/tests/networks/test_sdn_addresses.py:16]: Okay, got it. Thanks. Changed as you proposed.

File: cfme/tests/networks/test_sdn_subnets.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (mmojzis) [reply cfme/tests/networks/test_sdn_subnets.py:42]: Changed fstring.

**Comment by dajoRH** [comment]
Lint report for commit 5c64fb13e4cf54717f6760ed7c3f3e67a7bbc915:

`cfme/tests/networks/test_sdn_addresses.py`:
- :bangbang: Line 41:101: **E501** *line too long (102 > 100 characters)*

`cfme/tests/networks/test_sdn_subnets.py`:
- :bangbang: Line 41:101: **E501** *line too long (109 > 100 characters)*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 5c64fb13e4cf54717f6760ed7c3f3e67a7bbc915

The *local* fixture **`network_manager`** is used in the following files:
- **cfme/tests/networks/test_sdn_subnets.py**
    - *test_network_subnet_invalid_cidr*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 601268d3c76b734ffad16d12f4ae5e77c7a6fc3e

The *local* fixture **`network_manager`** is used in the following files:
- **cfme/tests/networks/test_sdn_subnets.py**
    - *test_network_subnet_invalid_cidr*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by john-dupuy** [comment]
@jaryn are you okay to move this one to 1LP?

File: cfme/networks/floating_ips.py [hunk 5]
```diff
@@ -53,6 +56,36 @@ def all(self):
...
+        """
+        view = navigate_to(self, 'Add')
+        view.fill({'network_manager': network_manager,
+                   'network': network_name,
+                   'network_port_id': network_port_id,
+                   'floating_ip_address': floating_ip_address,
+                   'fixed_ip_address': fixed_ip_address,
+                   'cloud_tenant': tenant})
+        view.add.click()
+        view.flash.assert_success_message(f'Floating IP "{floating_ip_address}" created')
```
Comment (jawatts) [inline cfme/networks/floating_ips.py:82]: We want to `assert_no error()` here instead and have the actual flash message asserted in a test.

File: cfme/tests/networks/test_sdn_addresses.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+pytestmark = [
+    test_requirements.sdn,
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([OpenStackProvider], scope="module")
+]
+
+
+@pytest.fixture()
+def network_manager(appliance, provider):
+    network_manager, = appliance.collections.network_providers.filter({"provider": provider}).all()
```
Comment (mshriver) [inline cfme/tests/networks/test_sdn_addresses.py:16]: Can you wrap this potential ValueError in a pytest.skip call, so it doesn't start throwing errors when there's more/less than 1 element in the list.

File: cfme/tests/networks/test_sdn_subnets.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (mshriver) [reply cfme/tests/networks/test_sdn_subnets.py:15]: @jaryn valid concern for sure.

I wouldn't consider this a blocker to merging this PR, but warrants some future enhancement.

This could be a globally available fixture (cfme/fixtures/), or even a subcollection of the parent cloud provider.

File: cfme/networks/floating_ips.py (hunk 5 above)
Comment (mshriver) [reply cfme/networks/floating_ips.py:82]: Agreed @jawatts, @mmojzis can you cover this flash assertion with a positive CRUD test case? Looks like only negative test cases are included in this PR.

File: cfme/tests/networks/test_sdn_addresses.py (hunk 6 above)
Comment (mmojzis) [reply cfme/tests/networks/test_sdn_addresses.py]: @mshriver Is this something like you wanted? https://github.com/ManageIQ/integration_tests/pull/10033/files#diff-d0ee99cb7213e3f233490166c76aab34R15

**Comment by mmojzis** [comment]
> Two small changes requested.
> 
> For the positive flash message, I'd accept this PR with just a `flash.assert_no_error` in the `create` method and no positive flash message text assertion.

@jawatts @mshriver 
These tests are only negative. It should check the format of flash message when invalid cidr is filled in.
I added these tests only because there was qe_test_coverage needed for these BZs.
I understand that you also want positive test that passes, but I didn't have qe_test_coverage for that.
I have no more time to write positive test.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 18e93fb79779d64ad9b29d3002a2ba8f8b72c645

The *local* fixture **`network_manager`** is used in the following files:
- **cfme/tests/networks/test_sdn_subnets.py**
    - *test_network_subnet_invalid_cidr*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:
